### PR TITLE
add a responsive degraded service banner to the pay page

### PR DESCRIPTION
**Pull request by Aierie**
On the smaller screen, the entire banner is a link and we only display the incident's name, similar to the mobile app. It's placed below the hamburger because having the hamburger's X and the hamburger button in the same place seems good - having the degraded service banner appear on top will push the hamburger down meaning that it would be lower than the X. On the large screen, we're conforming to the DApp's way of displaying a degraded service banner.

Yet to get design feedback. This aims to solve CS-3431 for the pay page.

small screen pay error page with a notice about degraded service below the top navbar.

large screen pay error page with a notice about degraded service above the top navbar, at the top of the screen.

### CODE REVIEW

**Comment by Aierie** [comment]
@oidoug wondering if you have some guidance for making the small screen version match the wallet?

File: packages/ssr-web/app/components/common/degraded-service-banner/index.css [hunk 1]
```diff
@@ -4,20 +4,17 @@
   --icon-background-color: var(--degraded-service-banner-text-color);
   --icon-color: var(--degraded-service-banner-background-color);
 
-  width: 100%;
   min-height: 2rem;
-  display: flex;
-  justify-content: center;
-  align-items: center;
```
Comment (backspace) [inline packages/ssr-web/app/components/common/degraded-service-banner/index.css:11]: The removal of this makes it look a bit too high for me; here’s me twiddling this on and off:

screencast 2022-03-25 10-17-35

Even with it on it looks maybe 1px too high for me but it’s less irksome at least 😆

Comment (backspace) [reply packages/ssr-web/app/components/common/degraded-service-banner/index.css:11]: oh ACTUALLY I was setting this on `…_container`

**Review by backspace — CHANGES_REQUESTED** [review]
This is a great addition, thanks. I have a sensitivity to vertical centring though 😆

File: packages/ssr-web/tests/acceptance/pay-test.ts [hunk 2]
```diff
@@ -719,4 +719,55 @@ module('Acceptance | pay', function (hooks) {
...
+          }
+        );
+      });
+
+      await visit(`/pay/sok`);
+      await waitFor(MERCHANT_MISSING_MESSAGE);
+
+      assert.dom('[data-test-degraded-service-banner]').isVisible({ count: 1 });
+
+      await percySnapshot(assert);
```
[the remark this answers is not in the document]
Comment (Aierie) [reply packages/ssr-web/tests/acceptance/pay-test.ts:770]: ya actually it doesn't matter if we only have one.

File: packages/ssr-web/app/components/common/degraded-service-banner/index.css (hunk 1 above)
Comment (Aierie) [reply packages/ssr-web/app/components/common/degraded-service-banner/index.css:11]: Ya vertical centering issues become more obvious with long text, thanks

**Review by backspace — APPROVED** [review]
🎉

File: packages/ssr-web/app/components/common/degraded-service-banner/small/index.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.degraded-service-banner--small {
+  border-radius: 8px;
```
Comment (oidoug) [inline packages/ssr-web/app/components/common/degraded-service-banner/small/index.css:2]: We're using 10 as radius in mobile.

**Review by oidoug — COMMENTED** [review]
Sorry I couldn't respond in a timely manner. Maybe my feedback is still valuable.

We've used three colors for these banners:
```
info: {
    iconColor: 'appleBlue',
    backgroundColor: '#F7F7F7',
    textColor: 'black',
  },
  warning: {
    iconColor: 'black',
    backgroundColor: '#F9D849',
    textColor: 'black',
  },
  error: {
    iconColor: 'black',
    backgroundColor: '#FF0000',
    textColor: 'white',
  },
```

Also, the "small" radius in mobile is always 10pts.

Screenshot_20220325-150838_Card Wallet

**Pull request merged**